src/item.cpp

```diff
@@ -10677,6 +10679,14 @@ int item::shots_remaining( const Character *carrier ) const
 int item::ammo_remaining( const std::set<ammotype> &ammo, const Character *carrier,
                           const bool include_linked ) const
 {
+    const bool is_tool_with_carrier = carrier != nullptr && type->tool;
```

Comment:
Using the same check as above.
```suggestion
    const bool is_tool_with_carrier = carrier != nullptr && is_tool();
```